lib/puppet_https.rb

```diff
@@ -20,19 +23,19 @@ def make_ssl_request(url, req)
     connection.read_timeout = @settings['read_timeout'] || 90 #A default timeout value in seconds
 
     connection.verify_mode = OpenSSL::SSL::VERIFY_PEER
-    ca_file = @settings['ca_certificate_path']
-    certpath = @settings['certificate_path']
+    ca_file   = @settings['ca_certificate_path']
+    certpath  = @settings['certificate_path']
     pkey_path = @settings['private_key_path']
 
-    if File.exists?(ca_file)
+    if ca_file and File.exists?(ca_file)
```

Comment:
Yes, this is to prevent an exception being thrown if the file paths were not provided. Because ca_file is nil if it's not being defined, the File.exists? will not be evaluated in that case: 

```
irb(main):001:0> ca_file = nil
=> nil
irb(main):002:0> if ca_file and File.exists?(ca_file)
irb(main):003:1> puts "nope"
irb(main):004:1> end
=> nil
```